src/transformState.js

```diff
@@ -5,7 +5,37 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const extraDataValue = action.extraData || {};
+    const typeOfAction = action.type;
+    const removeValue = action.keysToRemove || {};
```

Comment:
In order not to do unnecessary checks, you can not create variables for these objects, but directly use them where they can be.